### PR TITLE
Implement Clone, Debug for IntoIter

### PR DESCRIPTION
**Pull request by clarcharr**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -748,6 +748,18 @@ impl<A: Array> Drop for IntoIter<A> {
     }
 }
 
+impl<A: Array> Clone for IntoIter<A> where A::Item: Clone {
+    fn clone(&self) -> IntoIter<A> {
+        IntoIter { index: self.index, v: self.v.clone() }
```
Comment (bluss) [inline src/lib.rs]: We have potentially moved out from v, so we can't call clone here

Comment (clarcharr) [reply src/lib.rs]: Oh, whoops, thank you. Updated to a `.collect().into_iter()`

**Comment by clarcharr** [comment]
Fixed that, and added a rather naïve test to make sure that cloning doesn't drop anything more than it should.

**Comment by bluss** [comment]
Cool. Can we get that without the reformatting? Thanks. Sorry that I'll be away from maintaining a lot. So it's not a hurry.

**Comment by clarcharr** [comment]
Done! And it's totally fine; life happens.

**Comment by bluss** [comment]
Thanks!

**Pull request merged**